azurelinuxagent/common/version.py

```diff
@@ -116,6 +117,28 @@ def get_distro():
     return osinfo
 
 
+def get_lis_version():
+    """
+    This uses the Linux kernel's 'modinfo' command to retrieve the "version"
+    field for the "hv_vmbus" kernel module (the LIS drivers). This is the
+    documented method to retrieve the LIS version.
+    """
+    lis_version = "Unknown"
+    try:
+        command_output = shellutil.run_command(["modinfo", "-F", "version", "hv_vmbus"], log_error=True)
+        # If the system doesn't have LIS drivers, 'modinfo' will
+        # return nothing on stdout, which will cause 'run_command' to
+        # return an empty string.
+        if command_output:
+            lis_version = command_output
```

Comment:
we also need a more explicit message here. if we use empty then the log/telemetry will just include "LIS: "

many "not_installed"

Reply:
The current logic is, if we don't get a version string back from `modprobe`, we return "Unknown." So actually this will include `LIS: Unknown`.

Reply:
ok - not sure if you guys want to differentiate between unknown and no installed... if no need, ok as is

Reply:
Following up with you offline about this.

Reply:
While I've been unable to confirm if there's any case where the drivers would be absent on an Azure Linux VM, I've gone ahead and differentiated this anyway.